server/src/entities/system-config.entity.ts

```diff
@@ -1,115 +1,136 @@
 import { ConcurrentQueueName } from 'src/interfaces/job.interface';
 import { Column, Entity, PrimaryColumn } from 'typeorm';
 
+export type SystemConfigValue = string | string[] | number | boolean;
+
+// https://stackoverflow.com/a/47058976
+// https://stackoverflow.com/a/70692231
+type PathsToStringProps<T> = T extends SystemConfigValue
+  ? []
+  : {
+      [K in keyof T]: [K, ...PathsToStringProps<T[K]>];
+    }[keyof T];
+
+type Join<T extends string[], D extends string> = T extends []
+  ? never
+  : T extends [infer F]
+    ? F
+    : T extends [infer F, ...infer R]
+      ? F extends string
+        ? `${F}${D}${Join<Extract<R, string[]>, D>}`
+        : never
+      : string;
```

Comment:
What in the world. This is quite terrible lol. We should probably migrate to key value _per section_. Do like for the whole section json for a single key.

Reply:
I honestly don't know how it works except that it does lol